cupy/cuda/compiler.py

```diff
@@ -443,7 +443,10 @@ def _preprocess(source, options, arch, backend):
         raise ValueError('Invalid backend %s' % backend)
 
     assert isinstance(result, bytes)
-    return result
+
+    # Extract the part containing version information.
+    return '\n'.join(
+        x for x in result.decode().splitlines() if x.startswith('//'))
```

Comment:
Q: While this drops `.version`, `.target`, and `.address_size`, I assume this is OK because the former two are made up by other env keys, while the 3rd one is irrelevant?

Reply:
Thanks for the comment, Leo. I'm assuming these values do not change between CUDA bug-fix releases (NVRTC version = CUDA `major.minor` is already included as a cache key). Specifically for the `address_size`, the value seems to always 64 as all hosts supported by CUDA is 64-bit.

https://docs.nvidia.com/cuda/ptx-writers-guide-to-interoperability/index.html#fundamental-types